prov/rxd/src/rxd_init.c

```diff
@@ -57,14 +57,37 @@ static void rxd_init_env(void)
 	fi_param_get_int(&rxd_prov, "max_unacked", &rxd_env.max_unacked);
 }
 
+void rxd_info_to_core_mr_modes(uint32_t version, const struct fi_info *hints,
+			       struct fi_info *core_info)
+{
+	/* We handle FI_MR_BASIC and FI_MR_SCALABLE irrespective of version */
+	if (hints && hints->domain_attr &&
+	    (hints->domain_attr->mr_mode & (FI_MR_SCALABLE | FI_MR_BASIC))) {
+		core_info->mode = FI_LOCAL_MR;
+		core_info->domain_attr->mr_mode = hints->domain_attr->mr_mode;
+	} else if (FI_VERSION_LT(version, FI_VERSION(1, 5))) {
+		core_info->mode |= FI_LOCAL_MR;
+		/* Specify FI_MR_UNSPEC (instead of FI_MR_BASIC) so that
+		 * providers that support only FI_MR_SCALABLE aren't dropped */
+		core_info->domain_attr->mr_mode = FI_MR_UNSPEC;
+	} else {
+		core_info->domain_attr->mr_mode |= FI_MR_LOCAL;
+		if (!hints || !ofi_rma_target_allowed(hints->caps))
+			core_info->domain_attr->mr_mode |= OFI_MR_BASIC_MAP;
+		else if (hints->domain_attr)
+			core_info->domain_attr->mr_mode |=
+				hints->domain_attr->mr_mode & OFI_MR_BASIC_MAP;
+	}
+}
```

Comment:
This looks like an exact copy of the code in rxm.  Please move into a common area and avoid the code duplication.

Reply:
I knew you were going to say that!
I looked at it though and they have to be slightly different since rxm can support different mr modes whether or not RMA is requested. Changed it to reflect that but kept it separate because of the slight difference.